x-pack/dockerlogbeat/pipelinemanager/pipelineManager.go

```diff
@@ -103,12 +93,26 @@ func (pm *PipelineManager) CreateClientWithConfig(logOptsConfig map[string]strin
 //===================
 // Private methods
 
-// getPipeline gets a pipeline based on a confighash
-func (pm *PipelineManager) getPipeline(hashstring string) (*Pipeline, bool) {
+// checkAndCreatePipeline performs the pipeline check and creation as one atomic operation
+// It will either return a new pipeline, or an existing one from the pipeline map
+func (pm *PipelineManager) checkAndCreatePipeline(logOptsConfig map[string]string, file string) (*Pipeline, error) {
 	pm.mu.Lock()
 	defer pm.mu.Unlock()
-	pipeline, exists := pm.pipelines[hashstring]
-	return pipeline, exists
+
+	hashstring := makeConfigHash(logOptsConfig)
```

Comment:
The hash is more or less computes twice now. You already call `makeConfigHash` before this call.

Also compute the hash before acquiring the mutex. This reduces the time you need the mutex.